PR: Add support for remote development using Spyder's Remote client API

hlouzada opened the pull request:
Client side api for https://github.com/spyder-ide/envs-manager/pull/35

hlouzada committed:
feat: add api for envs_manager jupyter plugin using remote-client modules api

hlouzada committed:
feat: add api to env run action on remote server

hlouzada committed:
git subrepo pull --remote=https://github.com/hlouzada/envs-manager.git --branch=feat-jupyter-plugin --update --force external-deps/envs-manager

subrepo:
  subdir:   "external-deps/envs-manager"
  merged:   "89e3e7b"
upstream:
  origin:   "https://github.com/hlouzada/envs-manager.git"
  branch:   "feat-jupyter-plugin"
  commit:   "89e3e7b"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "cce3d93"

hlouzada committed:
fix: increase timeout for running manager actions

hlouzada committed:
feat: add proper error handling

hlouzada committed:
fix: add missing import

hlouzada committed:
rem: delete unused import

hlouzada committed:
git subrepo pull --remote=https://github.com/hlouzada/envs-manager.git --branch=feat-jupyter-plugin --update --force external-deps/envs-manager

subrepo:
  subdir:   "external-deps/envs-manager"
  merged:   "eb2b30e"
upstream:
  origin:   "https://github.com/hlouzada/envs-manager.git"
  branch:   "feat-jupyter-plugin"
  commit:   "eb2b30e"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "cce3d93"

hlouzada committed:
feat: add docker images to test the remote infrastructure

ccordoba12 commented on spyder_env_manager/spyder/workers.py:
```diff
@@ -102,3 +115,140 @@ def start(self):
             pass
+
+
+class RemoteEnvManagerApiError(SpyderRemoteAPIError):
```
Let's move the classes you added to this module to `spyder_env_manager/spyder/remoteclient.py` because it's about local operations only.

ccordoba12 reviewed (commented):
Thanks @hlouzada for implementing this so quickly! Besides my small suggestion, please address the style issues reported by our CI and install `pre-commit` to avoid them in the future.

hlouzada committed:
git subrepo pull --remote=https://github.com/hlouzada/envs-manager.git --branch=feat-jupyter-plugin --update --force external-deps/envs-manager

subrepo:
  subdir:   "external-deps/envs-manager"
  merged:   "21ed253"
upstream:
  origin:   "https://github.com/hlouzada/envs-manager.git"
  branch:   "feat-jupyter-plugin"
  commit:   "21ed253"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "cce3d93"

hlouzada committed:
fix: move remoteclient api modules to separate module

hlouzada committed:
refac: apply formatter

hlouzada committed:
git subrepo pull (merge) --remote=https://github.com/spyder-ide/envs-manager.git --branch=main --update --force external-deps/envs-manager

subrepo:
  subdir:   "external-deps/envs-manager"
  merged:   "37507ec"
upstream:
  origin:   "https://github.com/spyder-ide/envs-manager.git"
  branch:   "main"
  commit:   "37507ec"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "cce3d93"

hlouzada committed:
feat: add action to create enviroment spec

hlouzada committed:
refac: rename remote_id to server_id

ccordoba12 reviewed (approved):
Looks good to me now, thanks @hlouzada!
